paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/cdc/UpdatedDataFieldsProcessFunction.java

```diff
@@ -200,6 +213,29 @@ public static ConvertAction canConvert(DataType oldType, DataType newType) {
             return oldIdx <= newIdx ? ConvertAction.CONVERT : ConvertAction.IGNORE;
         }
 
+        oldIdx = DECIMAL_TYPES.indexOf(oldType.getTypeRoot());
```

Comment:
We don't need to support the decimal modification, we just need to determine if it is equal here.
```
if (oldType.equalsIgnoreNullable(newType)) {
    return ConvertAction.CONVERT;
}
```